cmd/ssl-selfsign-handshake-check/main.go

```diff
@@ -42,23 +30,79 @@ func init() {
 }
 
 func main() {
-	err := runSelfsignHandshake()
+	// create context
+	checkTimeLimit := time.Minute * 1
+	ctx, _ := context.WithTimeout(context.Background(), checkTimeLimit)
+
+	// create Kubernetes client
+	kubernetesClient, err := kubeClient.Create("")
+	if err != nil {
+		log.Errorln("Error creating kubeClient with error" + err.Error())
+	}
+
+	// hits kuberhealthy endpoint to see if node is ready
+	err = nodeCheck.WaitForKuberhealthy(ctx)
+	if err != nil {
+		log.Errorln("Error waiting for kuberhealthy endpoint to be contactable by checker pod with error:" + err.Error())
+	}
+
+	// fetches kube proxy to see if it is ready
+	err = nodeCheck.WaitForKubeProxy(ctx, kubernetesClient, "kuberhealthy", "kube-system")
+	if err != nil {
+		log.Errorln("Error waiting for kube proxy to be ready and running on the node with error:" + err.Error())
+	}
+	fileFound := whichCheck()
+	err = runCheck(fileFound)
 	if err != nil {
 		reportErr := reportKHFailure(err.Error())
 		if reportErr != nil {
 			log.Error(reportErr)
 		}
 		os.Exit(1)
 	}
-	reportErr := reportKHSuccess()
-	if reportErr != nil {
-		log.Error(reportErr)
+
+	if err == nil {
+		reportErr := reportKHSuccess()
+		if reportErr != nil {
+			log.Error(reportErr)
+		}
+		os.Exit(1)
+	}
+}
+
+// whichCheck determines if a cert has been provided, or if it needs to be retrieved from the host and returns a bool
+func whichCheck() bool {
```

Comment:
maybe rename `whichCheck` to `certFileProvided` or something more specific to what this function is doing?

Reply:
Updated func name.